redis_timeseries.go

```diff
@@ -0,0 +1,877 @@
+		}
+
+		if options.DuplicatePolicy != "" {
+			args = append(args, "DUPLICATE_POLICY", options.DuplicatePolicy)
+		}
+		if options.Labels != nil {
+			args = append(args, "LABELS")
+			for label, value := range options.Labels {
+				args = append(args, label, value)
+
+			}
+		}
+	}
+	cmd := NewIntCmd(ctx, args...)
+	_ = c(ctx, cmd)
+	return cmd
+}
+
+// TSCreate - Creates a new time-series key.
+// For more information - https://redis.io/commands/ts.create/
+func (c cmdable) TSCreate(ctx context.Context, key string) *StatusCmd {
+	args := []interface{}{"TS.CREATE", key}
+	cmd := NewStatusCmd(ctx, args...)
+	_ = c(ctx, cmd)
+	return cmd
+}
+
+// TSCreateArgs - Creates a new time-series key with additional options.
+// This function allows for specifying additional options such as:
+// Retention, ChunkSize, Encoding, DuplicatePolicy and Labels.
+// For more information - https://redis.io/commands/ts.create/
+func (c cmdable) TSCreateArgs(ctx context.Context, key string, options *TSOptions) *StatusCmd {
+	args := []interface{}{"TS.CREATE", key}
+	if options != nil {
+		if options.Retention != 0 {
+			args = append(args, "RETENTION", options.Retention)
+		}
+		if options.ChunkSize != 0 {
+			args = append(args, "CHUNK_SIZE", options.ChunkSize)
+
+		}
+		if options.Encoding != "" {
+			args = append(args, "ENCODING", options.Encoding)
+		}
+
+		if options.DuplicatePolicy != "" {
+			args = append(args, "DUPLICATE_POLICY", options.DuplicatePolicy)
+		}
+		if options.Labels != nil {
+			args = append(args, "LABELS")
+			for label, value := range options.Labels {
+				args = append(args, label, value)
+
+			}
+		}
+	}
+	cmd := NewStatusCmd(ctx, args...)
+	_ = c(ctx, cmd)
+	return cmd
+}
+
+// TSAlterArgs - Alters an existing time-series key with additional options.
+// This function allows for specifying additional options such as:
+// Retention, ChunkSize and DuplicatePolicy.
+// For more information - https://redis.io/commands/ts.alter/
+func (c cmdable) TSAlterArgs(ctx context.Context, key string, options *TSAlterOptions) *StatusCmd {
+	args := []interface{}{"TS.ALTER", key}
+	if options != nil {
+		if options.Retention != 0 {
+			args = append(args, "RETENTION", options.Retention)
+		}
+		if options.ChunkSize != 0 {
+			args = append(args, "CHUNK_SIZE", options.ChunkSize)
+
+		}
+		if options.DuplicatePolicy != "" {
+			args = append(args, "DUPLICATE_POLICY", options.DuplicatePolicy)
+		}
+		if options.Labels != nil {
+			args = append(args, "LABELS")
+			for label, value := range options.Labels {
+				args = append(args, label, value)
+
+			}
+		}
+	}
+	cmd := NewStatusCmd(ctx, args...)
+	_ = c(ctx, cmd)
+	return cmd
+}
+
+// TSCreateRule - Creates a compaction rule from sourceKey to destKey.
+// For more information - https://redis.io/commands/ts.createrule/
+func (c cmdable) TSCreateRule(ctx context.Context, sourceKey string, destKey string, aggregator string, bucketDuration int) *StatusCmd {
```

Comment:
does it make more sense, in the library to be a supported enum instead - using the [iota enum pattern](https://golangbyexample.com/iota-in-golang/)? You'd have to ensure those are available, but it's IMHO better than a string with docs. WDYT?